src/aiida_quantumespresso/parsers/parse_raw/ph.py

```diff
@@ -236,6 +276,32 @@ def detect_important_message(logs, line):
             except Exception:
                 pass
 
+        elif 'Diagonalizing the dynamical matrix' in line:
+
+            mode_count = count
+
+            while not 'Calculation' in lines[mode_count] and not 'CPU' in lines[mode_count]:
+                mode_count += 1
+
+            symlabel_q_point, q_data = parse_mode_symmetries(lines[count: mode_count], num_atoms)
+
+            for q_index, q_point in parsed_data['q_points'].items():
+                if numpy.isclose(
+                    numpy.array(q_point), numpy.array(symlabel_q_point), rtol=0, atol=1e-7
+                ).all():
+                    parsed_data.setdefault('symmetry_labels', {})
+                    parsed_data['symmetry_labels'][q_index] = q_data
+
+            parsed_data['num_q_found'] += 1
+
+    # Remove the q-points from the parsed data; these are only used to assign the symmetry labels to the right index
+    parsed_data.pop('q_points', None)
+
+    # Trim the number of irreps to the number of q-points finished in this run
+    if parsed_data['num_q_found'] > 0:
+        parsed_data['number_of_irr_representations_for_each_q'] = \
+            parsed_data['number_of_irr_representations_for_each_q'][:parsed_data['num_q_found']]
+
```

Comment:
In case no q-points are found to parse (should fix the duplication in 'number_or_irr_representations_for_each_q' in the merged output mentioned in previous comment)

```suggestion
    if parsed_data['num_q_found'] >= 0:
        parsed_data['number_of_irr_representations_for_each_q'] = \
            parsed_data['number_of_irr_representations_for_each_q'][:parsed_data['num_q_found']]
 
```

Reply:
should fix:
#
> When a PhBaseCalculation has multiple PhCalcuations and the wall-time is hit mid q-point calculation there is an extra entry in `number_of_irr_representations_for_each_q` as there are two 'headers' for the same q-point in two different `PhCalculations` - This is probably also due to an error in the parser

Reply:
This unfortunately breaks one of my tests 😅. Could you provide some example output files for this case? Then I can also add a test for it and see how I can make sure both run successfully.

Reply:
Never mind! I generated a simple test case myself. But it's clear the parsing isn't giving us the desired outputs:

```yaml
dynamical_matrix_0:
  frequencies:
  - -3.151844
  - -3.151844
  - -3.139229
  frequencies_units: cm-1
  q_point:
  - 0.0
  - 0.0
  - 0.0
  q_point_units: 2pi/lattice_parameter
number_of_atoms: 1
number_of_irr_representations_for_each_q:
- 2
- 2
number_of_qpoints: 1
wall_time_seconds: 27.53

```

Reply:
The basic idea was solid, but I propose an alternative solution in [0c5cc84](https://github.com/aiidateam/aiida-quantumespresso/pull/818/commits/0c5cc8449c4c4d412bc291a2cdc7c88d11473cff), which basically boils down to the same thing but doesn't break for cases where the `ph.x` calculation didn't print the irr repr line at all. Now the outputs are starting to look pretty good:

```yaml
dynamical_matrix_1:
  frequencies:
  - -3.151844
  - -3.151844
  - -3.139229
  frequencies_units: cm-1
  mode_symmetry:
  - L_2'
  - L_3'
  - L_3'
  point_group: D_3d (-3m)
  q_point:
  - 0.0
  - 0.0
  - 0.0
  q_point_units: 2pi/lattice_parameter
number_of_atoms: 1
number_of_irr_representations_for_each_q:
- 2
number_of_qpoints: 1
wall_time_seconds: 27.53

```